parts/kubernetesmastercustomdata.yml

```diff
@@ -335,7 +338,57 @@ write_files:
     fi
     mount $MOUNTPOINT
 
+{{if .OrchestratorProfile.IsCustomEtcdVersion}}
+- path: "/etc/systemd/system/etcd.service"
+  permissions: "0644"
+  owner: "root"
+  content: |
+    [Unit]
+    Description=etcd - highly-available key value store
+    Documentation=https://github.com/coreos/etcd
+    Documentation=man:etcd
+    After=network.target
+    Wants=network-online.target
+    [Service]
+    Environment=DAEMON_ARGS=
+    Environment=ETCD_NAME=%H
+    Environment=ETCD_DATA_DIR=/var/lib/etcd/default
+    EnvironmentFile=-/etc/default/%p
+    Type=notify
+    User=etcd
+    PermissionsStartOnly=true
+    ExecStart=/usr/bin/etcd $DAEMON_ARGS
+    Restart=on-abnormal
+    [Install]
+    WantedBy=multi-user.target
+    Alias=etcd.service
+
+- path: "/opt/azure/containers/setup-etcd.sh"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    #!/bin/bash
+    # if provisioning using any etcd version other than 2.5.2
+    # We download it and set it up here 
+    set -x
+    ETCD_VER=v{{ .OrchestratorProfile.KubernetesConfig.EtcdVersion }}
+    DOWNLOAD_URL=https://github.com/coreos/etcd/releases/download # we default to github
```

Comment:
this should go through CDN (Dmitry can show you) so that this is not down if github goes down.